### PR TITLE
Enable fsspec by default for cli config file

### LINKED ISSUE #7360: LightningCLI: Support config files on object stores
## 🚀 Feature

In addition to `python trainer.py --config config.yaml`, LightningCLI should enable `python trainer.py --config s3://bucket/config.yaml` 

### Motivation

pytorch-lightning has good support for remote filesystems thanks to #2164 #3320 (and possibly others). Thanks to that, it's straightforward to deploy lightning-based scripts to compute cluster environments without having to write extra code preparing / post-processing the local file-systems https://pytorch-lightning.readthedocs.io/en/latest/clouds/cluster.html It would be nice to extend this to LightningCLI.

When using LightningCLI however, it's currently necessary to deploy the config.yaml file to each node, instead of referencing a copy in an object store such as S3.

cc @mauvilsa?

### PR DESCRIPTION
**Pull request by mauvilsa**
## What does this PR do?

Fixes #7360.

## Before submitting
- [X] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [X] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [X] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [X] Is this pull request ready for review? (if not, please submit in draft mode)
 - [X] Check that all items from **Before submitting** are resolved
 - [X] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/utilities/cli.py [hunk 1]
```diff
@@ -24,10 +24,12 @@
 
 _JSONARGPARSE_AVAILABLE = _module_available("jsonargparse")
 if _JSONARGPARSE_AVAILABLE:
-    from jsonargparse import ActionConfigFile, ArgumentParser
+    from jsonargparse import ActionConfigFile, ArgumentParser, set_config_read_mode
 else:
     ArgumentParser = object
 
+set_config_read_mode(fsspec_enabled=True)
+
```
Comment (carmocca) [inline pytorch_lightning/utilities/cli.py:32]: This should be called inside `if _JSONARGPARSE_AVAILABLE:`

**Review by carmocca — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -10,6 +10,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 ### Added
 
 
+- Added LightningCLI support for config files on object stores ([#7521](https://github.com/PyTorchLightning/pytorch-lightning/pull/7430))
```
Comment (awaelchli) [inline CHANGELOG.md:13]:
```suggestion
- Added LightningCLI support for config files on object stores ([#7521](https://github.com/PyTorchLightning/pytorch-lightning/pull/7521))
```

**Review by awaelchli — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (awaelchli) [reply CHANGELOG.md:13]: 🦅 👁️

**Review by tchaton — APPROVED** [review]
LGTM !

**Pull request merged**